connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Herder.java

```diff
@@ -127,6 +129,21 @@
      */
     ConnectorStateInfo.TaskState taskStatus(ConnectorTaskId id);
 
+    /**
+     * Get the configuration definitions for the connector class.
+     * @param connType the connector class
+     * @param callback callback to invoke upon completion
+     */
+    void getConfigDef(String connType, Callback<ConfigDefInfo> callback);
```

Comment:
Kafka doesn't use `getX`.

Also, I think I asked this about the status API too -- is there anything async here that requires this to provide a callback? Can it just return the `ConfigDefInfo` synchronously?

Related to the question about new `Connector` APIs, do we need both `getConfigDef` and `validateConfigs`? Why not just the one, where an empty config returns the relevant `ConfigInfos` given an empty connector configuration? Providing both REST endpoints might be handy (i.e. the `GET` vs `POST`), although even that I am not convinced of. (For example, connector configs should always include a connector name anyway; should we require that when these connector configs are validated?)